### PR TITLE
PPL Lookup

### PR DESCRIPTION
**Pull request by salyh**
### Description
Implement PPL Lookup

### Issues Resolved
https://github.com/opensearch-project/sql/issues/2651

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by YANG-DB** [comment]
@salyh please take a look at this existing [PPL correlation API ](https://github.com/opensearch-project/opensearch-spark/blob/main/docs/PPL-Correlation-command.md)

**Comment by salyh** [comment]
@YANG-DB any comments on the proposal so far?

**Comment by salyh** [comment]
@rupal-bq @anasalkouz any comments on the proposal so far?

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java [hunk 1]
```diff
@@ -256,6 +267,93 @@ public LogicalPlan visitDedupe(Dedupe node, CatalystPlanContext context) {
...
+
+        while(!root.getChild().isEmpty()) {
+            root = root.getChild().get(0);
+        }
+
+        org.opensearch.sql.ast.tree.Relation source = (org.opensearch.sql.ast.tree.Relation) root;
+
+        node.getChild().get(0).accept(this, context);
+
+        //TODO: not sure how to implement appendonly
```
Comment (LantaoJin) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:282]: To implement `appendonly`, the output Project (parent of Join) should be
```
Project [other fields in source, CASE WHEN isnull('source.copyField) THEN coalesce('source.copyField, 'lookup.copyField) AS copyField ELSE 'lookup.copyField END AS copyField]
  +- Join LeftOuter
```
In `CASE WHEN expression1 ELSE expression2`, the expression1 is for appendonly (not performing when the source field already exist), the else is for overwriting existing field.

Comment (kt-eliatra) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:282]: Doesn't it mean that we need to know list of fields available in the "source" table, in order to build project consisting of two lists:
a list of fields which we want to overwrite with values from the lookup table and a list of fields which should remain untouched?
I'm not sure whether it is possible to retrieve such list of available fields.

**Comment by LantaoJin** [comment]
Close this as issue resolved by https://github.com/opensearch-project/opensearch-spark/pull/686

**Pull request closed without merging**